tests/testthat/test-tbl-df.R

```diff
@@ -36,6 +36,22 @@ test_that("[ with 0 cols creates correct row names (#656)", {
   expect_identical(zero_row, as_tibble(iris)[0])
 })
 
+test_that("[ with 0 cols returns correct number of rows", {
+  iris_tbl <- as_tibble(iris)
+  nrow_iris <- nrow(iris_tbl)
+
+  expect_is(iris_tbl, "tbl_df")
```

Comment:
Well I just imitated the structure from previous `test_that`. This `expect_is` ensures that we test exactly the `[.tbl_df` function in current test. I actually can remove it because such check is made in previous test but I think this way fills more bulletproof. Of course it is up to you to decide.